pkg/controller/core/workload_controller.go

```diff
@@ -425,3 +443,51 @@ func handlePodOverhead(log logr.Logger, wl *kueue.Workload, c client.Client) {
 		}
 	}
 }
+
+type limitRangeHandler struct {
+	client client.Client
+}
+
+func (h *limitRangeHandler) Create(e event.CreateEvent, _ workqueue.RateLimitingInterface) {
+	h.deleteWorkloads(e.Object)
+}
+
+func (h *limitRangeHandler) Update(e event.UpdateEvent, _ workqueue.RateLimitingInterface) {
+	h.deleteWorkloads(e.ObjectNew)
+}
+
+func (h *limitRangeHandler) Delete(e event.DeleteEvent, _ workqueue.RateLimitingInterface) {
+	h.deleteWorkloads(e.Object)
+}
+
+func (h *limitRangeHandler) Generic(_ event.GenericEvent, _ workqueue.RateLimitingInterface) {
+}
+
+func (h *limitRangeHandler) deleteWorkloads(obj client.Object) {
+	lr, ok := obj.(*corev1.LimitRange)
+	if !ok {
+		return
+	}
+
+	//TODO: the eventHandler should get a context soon, and this could be dropped
+	// https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/handler/eventhandler.go
+	ctx := context.TODO()
+	log := ctrl.LoggerFrom(ctx).WithValues("limitRange", klog.KObj(lr))
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	lst := kueue.WorkloadList{}
+	err := h.client.List(ctx, &lst, client.InNamespace(lr.Namespace), client.MatchingFields{indexer.WorkloadAdmittedKey: string(metav1.ConditionFalse)})
+	if err != nil {
+		log.Error(err, "Could not list the unatmitted workloads")
+	}
+	log.V(5).Info("deleting workloads", "count", len(lst.Items))
+
+	for _, w := range lst.Items {
+		log.V(5).Info("deleting", "workload", klog.KObj(&w))
```

Comment:
Better move it after the Delete call

```suggestion
		log.V(2).Info("Deleted Workload due to changing LimitRange", "workload", klog.KObj(&w))
```